sale_automatic_workflow_payment_mode/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,64 @@
+# © 2016 Camptocamp SA, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+from odoo.addons.sale_automatic_workflow.models.automatic_workflow_job import savepoint
+
+_logger = logging.getLogger(__name__)
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    @api.model
+    def run_with_workflow(self, sale_wkf):
+        workflow_domain = [("workflow_process_id", "=", sale_wkf.id)]
+        res = super(AutomaticWorkflowJob, self).run_with_workflow(sale_wkf)
+        if sale_wkf.register_payment:
```

Comment:
I don't get how this is related with `payment_mode`. This is something standard...

Reply:
IMO, it is not necessary to add anything extra here since for each payment mode a `sale.workflow.process` is defined, and therefore in the onchange of the `payment_mode_id` of the order, you have already defined the correct workflow; therefore, when this action is executed, it is no longer necessary to take into account the payment mode.

Reply:
The question is that this is registering a payment, which is something standard, not linked with `payment.mode`. Why is this code here? I understand this should go in the main module if needed, and if it's modifying a bit such registering, then it should be done by a hook on main module + the extension method here.

Reply:
Ok, I understand.
I have added the `_register_payments()` function to the `sale_automatic_workflow` and also, in `sale_automatic_workflow_payment_mode` I have added only the necessary related to the payment mode.